megamek/src/megamek/common/MekFileParser.java

```diff
@@ -151,7 +153,18 @@ public Entity getEntity() {
         return m_entity;
     }
 
-    public void parse(InputStream is, String fileName) throws Exception {
+    public void parse(InputStream ns, String fileName) throws Exception {
+        File tempFile = File.createTempFile("temp", ".tmp");
+        tempFile.deleteOnExit();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(ns));
+             BufferedWriter writer = new BufferedWriter(new FileWriter(tempFile))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                writer.write(line);
+                writer.newLine();
+            }
+        }
+        InputStream is = new FileInputStream(tempFile);
```

Comment:
I hate it as much as everybody else... best way would be to move this to its own small method and nobody talks about it ever again